src/scenarios/billing/SubscribeToSubscriptionPlan/SubscribeToSubscriptionPlan.tsx

```diff
@@ -0,0 +1,476 @@
+import {
+  useScenarioHost,
+  VariableSearch,
+} from '@axinom/mosaic-fe-samples-host';
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  Container,
+  Divider,
+  Form,
+  Header,
+  Label,
+  Segment,
+} from 'semantic-ui-react';
+import { getApolloClient } from '../../../apollo-client';
+import {
+  getPaymentPlanPricesQuery,
+  subscribeRedirectMutation,
+  getPaymentPlansQuery,
+  subscribePopupMutation,
+  activateSubscriptionMutation,
+} from './graphql-documents';
+import { loadScript } from '@paypal/paypal-js';
+
+interface PaymentPlan {
+  id: string;
+  title: string;
+  periodQuantity: number;
+  periodUnit: string;
+  subscriptionPlan: {
+    title: string;
+  };
+}
+
+interface PaymentPlanPrice {
+  country: string;
+  currency: string;
+  price: string;
+}
+
+type PurchaseFlow = 'POPUP' | 'REDIRECT';
+
+export const SubscribeToSubscriptionPlan: React.FC = () => {
+  const { activeProfile, logger } = useScenarioHost();
+  const [userAccessToken, setUserAccessToken] = useState<string>('');
+  const [paypalClientId, setPaypalClientId] = useState<string>('');
+  const [paymentPlans, setPaymentPlans] = useState<PaymentPlan[]>([]);
+  const [paymentPlanId, setPaymentPlanId] = useState<string>();
+  const [paymentPlanPrices, setPaymentPlanPrices] = useState<
+    PaymentPlanPrice[]
+  >([]);
+  const [country, setCountry] = useState<string>();
+  const [purchaseFlow, setPurchaseFlow] = useState<PurchaseFlow>('REDIRECT');
+  const [paypalInitializing, setPaypalInitializing] = useState<boolean>(false);
+
+  const apolloClient = getApolloClient(
+    new URL('graphql', activeProfile.billingServiceBaseURL).href,
+  );
+
+  useEffect(() => {
+    const pathname = window.location.pathname;
+    const urlParams = new URLSearchParams(window.location.search);
+    const subscriptionId = urlParams.get('subscriptionId');
+
+    if (subscriptionId) {
+      logger.log('Redirection from Paypal detected.');
+      if (pathname.endsWith('subscribe-success')) {
+        logger.log(
+          `Subscription with ID ${subscriptionId} should be active now. You can verify it by using 'List Subscription Plans' scenario.`,
+        );
+      } else if (pathname.endsWith('subscribe-error')) {
+        logger.log(
+          `An error occurred in subscription with ID ${subscriptionId}.`,
+        );
+      }
+    }
+  }, [logger]);
+
+  const fetchPaymentPlans = async (): Promise<void> => {
```

Comment:
the function (and implicitly the logs - when we use `func.name` in the log string-interpolcation) can be renamed to `fetchPaymentPlansAndPayPalSettings` - as we're fetching those two things